src/components/ecl-button/ecl-button.tsx

```diff
@@ -37,42 +43,69 @@ export class EclButton {
 
     if (this.el.getElementsByTagName('ecl-icon')[0] && this.el.querySelector('.ecl-icon')) {
       const slot = this.el.getElementsByTagName('ecl-icon')[0].getAttribute('slot');
-      this.el.querySelector('.ecl-icon').classList.add('ecl-button__icon');
+      this.el.querySelector('.ecl-icon').classList.add('ecl-button__icon', `sc-ecl-button-${this.theme}`);
       if (slot && !this.hideLabel) {
-        this.el.querySelector('.ecl-icon').classList.add(`ecl-button__${slot.substring(0, 5) + '-' + slot.substring(5)}`, `sc-ecl-button-${this.theme}`);
+        this.el.querySelector('.ecl-icon').classList.add(`ecl-button__${slot.substring(0, 5) + '-' + slot.substring(5)}`);
       }
     }
+
+    const iconBefore = this.el.querySelector('[slot="icon-before"]');
+    const iconAfter = this.el.querySelector('[slot="icon-after"]');
+
+    this.hasIconBefore = !!iconBefore;
+    this.hasIconAfter = !!iconAfter;
   }
 
   getClass(): string {
     return [
-      `ecl-button`,
+      'ecl-button',
       `ecl-button--${this.variant}`,
-      this.styleClass
-    ].join(' ');
+      this.hideLabel ? 'ecl-button--icon-only' : '',
+      this.styleClass,
+    ].join(' ').trim();
   }
 
   render() {
     return (
       <button
         class={this.getClass()}
         type={this.type}
+        disabled={this.disabled}
         {...(this.ariaControls && { 'aria-controls': this.ariaControls })}
         {...(this.itemId && { id: this.itemId })}
       >
         <span class="ecl-button__container">
-          <slot name="icon-before"></slot>
-        { !this.hideLabel ?
-          <span class="ecl-button__label">
-            <slot></slot>
-          </span> : 
-          <span class="ecl-u-sr-only" data-ecl-label>
-            <slot></slot>
-          </span>
-        }
-          <slot name="icon-after"></slot>
+          {this.hasIconBefore && !this.indicator && (
+            <slot name="icon-before"></slot>
+          )}
+          {this.indicator && this.hasIconBefore && (
+            <span class="ecl-button__icon-container">
+              <slot name="icon-before"></slot>
+              <ecl-indicator value={this.indicatorValue}></ecl-indicator>
```

Comment:
we only load the js of a component on request, you need to add an attribute, ecl-script for this to happen

Reply:
added, i am a bit uncertain to add it in ecl-button component because i saw the other examples, the ecl-script is added in the stories.